src/app/[lng]/earn/page.tsx

```diff
@@ -11,10 +13,16 @@ export const metadata: Metadata = {
   },
 };
 
-export default async function Page() {
+interface PageProps {
+  searchParams: Promise<EarnsPageSearchParams>;
+}
+
+export default async function Page(props: PageProps) {
+  const searchParams = await props.searchParams;
+  const initialFilters = parseFiltersFromUrl(searchParams);
```

Comment:
To me that's leaking internals of the filtering context, it should be the only one manipulating filtering related data. Could you keep it inside the filtering context?